File: java/sprykerFileCreator/ui/ClassSelectionModel.java
package sprykerFileCreator.ui;

import javax.swing.table.AbstractTableModel;
import java.util.List;

public class ClassSelectionModel extends AbstractTableModel {

    private List<ClassSelectionItem> items;

    public ClassSelectionModel(List<ClassSelectionItem> items) {
        this.items = items;
    }

    @Override
    public int getRowCount() {
        return this.items.size();
    }

    @Override
    public int getColumnCount() {
        return 3;
    }

    @Override
    public Object getValueAt(int rowIndex, int columnIndex) {
        ClassSelectionItem item = items.get(rowIndex);

        switch(columnIndex){
            case 0: return item.isSelected();
            case 1: return item.getClassTypeLabel();
            case 2: return item.getClassName();
            default: return "";
        }
    }

    @Override
    public boolean isCellEditable(int rowIndex, int columnIndex) {
        switch(columnIndex){
            case 0: return true;
            case 1: return false;
            case 2: return items.get(rowIndex).isClassNameEditable();
            default: return false;
        }
    }

    @Override
    public Class<?> getColumnClass(int column) {

        if(column == 0){
            return Boolean.class;
        }

        if(column == 2){
            return String.class;
        }

        return super.getColumnClass(column);
    }

    @Override
    public void setValueAt(Object aValue, int rowIndex, int columnIndex) {
        ClassSelectionItem item = items.get(rowIndex);

        if (columnIndex == 0 && aValue instanceof Boolean) {
            Boolean isSelected = (Boolean) aValue;
            item.setSelected(isSelected);

        }

        if (columnIndex == 2 && aValue instanceof String) {
            String className = (String) aValue;
            item.setClassName(className);
        }

        fireTableCellUpdated(rowIndex, columnIndex);
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/ui/ClassSelectionModel.kt
package com.github.tobi812.sprykerplugin.ui

import com.github.tobi812.sprykerplugin.ui.ClassSelectionItem.isSelected
import com.github.tobi812.sprykerplugin.ui.ClassSelectionItem.classTypeLabel
import com.github.tobi812.sprykerplugin.ui.ClassSelectionItem.className
import com.github.tobi812.sprykerplugin.ui.ClassSelectionItem.isClassNameEditable
import javax.swing.table.AbstractTableModel

class ClassSelectionModel(private val items: List<ClassSelectionItem>) : AbstractTableModel() {
    override fun getRowCount(): Int {
        return items.size
    }

    override fun getColumnCount(): Int {
        return 3
    }

    override fun getValueAt(rowIndex: Int, columnIndex: Int): Any {
        val item = items[rowIndex]
        return when (columnIndex) {
            0 -> item.isSelected!!
            1 -> item.classTypeLabel!!
            2 -> item.className
            else -> ""
        }
    }

    override fun isCellEditable(rowIndex: Int, columnIndex: Int): Boolean {
        return when (columnIndex) {
            0 -> true
            1 -> false
            2 -> items[rowIndex].isClassNameEditable!!
            else -> false
        }
    }

    override fun getColumnClass(column: Int): Class<*> {
        if (column == 0) {
            return Boolean::class.java
        }
        return if (column == 2) {
            String::class.java
        } else super.getColumnClass(column)
    }

    override fun setValueAt(aValue: Any, rowIndex: Int, columnIndex: Int) {
        val item = items[rowIndex]
        if (columnIndex == 0 && aValue is Boolean) {
            item.isSelected = aValue
        }
        if (columnIndex == 2 && aValue is String) {
            item.className = aValue
        }
        fireTableCellUpdated(rowIndex, columnIndex)
    }
}